packages/discovery/src/discovery/provider/DiscoveryProvider.test.ts

```diff
@@ -0,0 +1,152 @@
+import { expect, mockFn, MockObject, mockObject } from 'earl'
+import { providers } from 'ethers'
+
+import { EthereumAddress } from '../../utils/EthereumAddress'
+import { EtherscanLikeClient } from '../../utils/EtherscanLikeClient'
+import { DiscoveryLogger } from '../DiscoveryLogger'
+import { DiscoveryProvider } from './DiscoveryProvider'
+
+const rangesFromCalls = (provider: MockObject<providers.Provider>) =>
+  provider.getLogs.calls.map((call) => [
+    call.args[0].fromBlock,
+    call.args[0].toBlock,
+  ])
+
+const GETLOGS_MAX_RANGE = 10000
+
+describe(DiscoveryProvider.name, () => {
+  describe(DiscoveryProvider.prototype.getLogs.name, () => {
+    const etherscanLikeClientMock = mockObject<EtherscanLikeClient>({})
+    const address = EthereumAddress.random()
+    const topics = ['testTopic']
+    let providerMock: MockObject<providers.Provider>
+    let discoveryProviderMock: DiscoveryProvider
```

Comment:
Can't this be just? Why does it have to be in the beforeEach?

```typescript
const providerMock = mockObject<providers.Provider>({
    getLogs: mockFn().resolvesTo([]),
})
const discoveryProviderMock = new DiscoveryProvider(
    providerMock,
    etherscanLikeClientMock,
    DiscoveryLogger.SILENT,
    GETLOGS_MAX_RANGE,
)
```

Reply:
In tests I check `providerMock.getLogs.calls`, so we need to create a fresh one for each new test.